custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -138,6 +149,131 @@ def handle_exception(self, exception, repeat_record):
             _save_error_message(repeat_record.domain, repeat_record.payload_id, unicode(exception))
 
 
+class NikshayFollowupPayloadGenerator(BasePayloadGenerator):
+    @property
+    def content_type(self):
+        return 'application/json'
+
+    def get_payload(self, repeat_record, test_case):
+        """
+        "PatientID": "Nikshay-ID",
+        "TestDate": "dd/mm/yyyy",
+        "LabNo": "23",
+        "RegBy": "nikshay-user",
+        "Local_ID": "patients-case-id",
+        "IP_From": "Servers IP address or Mobile device MAC Address",
+        "IntervalId": 0:4,
+        "PatientWeight": 45,
+        "SmearResult": 0:99,
+        "Source": nikshay-id,
+        "DMC": owner_id
+
+        :param repeat_record:
+        :param episode_case:
+        :return:
+        """
+        lab_referral_case = get_lab_referral_from_test(test_case.get_id)
+        dmc = SQLLocation.active_objects.get(location_id=lab_referral_case.owner_id)
+        dmc_code = dmc.metadata.get('nikshay_code')
+        episode_case = get_episode_case_from_test(test_case.get_id)
+        person_case = get_person_case_from_episode(episode_case.get_id)
+
+        test_case_properties = test_case.dynamic_case_properties()
+        episode_case_properties = episode_case.dynamic_case_properties()
+
+        if test_case_properties.get('purpose_of_testing') == 'diagnostic':
+            interval_id = 0
+        else:
+            interval_id = purpose_of_testing.get(test_case_properties.get('follow_up_test_reason'))
+
+        properties_dict = {
+            "PatientID": episode_case_properties.get('nikshay_id'),
+            "TestDate": datetime.datetime.strptime(test_case_properties.get('date_tested'),
+                                                   '%Y-%m-%d').strftime('%d/%m/%Y'),
+            # MK to test if there is a limit of length of this number and truncate for that accordingly
+            "LabNo": test_case_properties.get('lab_serial_number', 0),
```

Comment:
@proteusvacuum Tested this for length constraints for length 5 as highlighted [here](https://docs.google.com/spreadsheets/d/1AdhiHTn9Z47VCGIKXCYL-UXDvV9-aV1d8gRzVYVEh94/edit#gid=609100483). Could submit a lab number `234567891078`, so seems like a false restriction.